src/main/java/com/upserve/uppend/lookup/LookupData.java

```diff
@@ -582,7 +599,7 @@ protected LookupMetadata getMetadata() {
                 boolean luckyMe = reloadStamp.compareAndSet(stamp[0], stamp[0] + reloadInterval);
 
                 if (luckyMe) {
-                    result = loadMetadata();
+                    result = loadMetadata(result);
                     timeStampedMetadata.set(result, stamp[0] + reloadInterval);
```

Comment:
The intent was that if reloadInterval == 0 
There is no reload - it is deactivated. Let's confirm that is working correctly - not reloading every time instead!

Reply:
If `reloadInterval` is 0, then the `compareAndSet` returns `true` every time -- this causes a reload _every_ time the `getMetaData` method is called.

I've inverted the logic and it works properly now.

Reply:
Well that should be a big improvement.